### PR TITLE
Global and local safe-scaling of the components 

### PR DESCRIPTION
**Pull request by Rmano**
This is quite big and unfinished. I'll copy and paste from the manual to give a feeling... For now, I have managed to adjust resistors, capacitors, inductors, and diodes. I think I'll do generators, transistors, and tubes too, at least. Maybe mechanical analogy and block thingies.  

image
image

image

image

Related to #218

### CODE REVIEW

File: doc/circuitikzmanual.tex [hunk 1]
```diff
@@ -1351,6 +1325,191 @@ \subsection{DC sources}
...
+        node[oscopeshape, fill=yellow!20!orange](O2){$C_2$};
+    \draw (O2.in 2) to[short, *-] ++(0,-0.5) node[ground]{};
+    \draw (O2.in 1) to[short, *-] ++(0,-0.5)
+           -- ++(-1,0) node[currarrow, xscale=-1]{};
+\end{circuitikz}
+\end{LTXexample}
+
+\subsubsection{Measuring voltage and currents, multiple ways}
+
+This is the classical (legacy) option, with the \texttt{voltmeter} and \texttt{ammeter}. The problem is that elements are intrinsically horizontal and so they looks funny if put in vertical way.
```
Comment (asinghvi17) [inline doc/circuitikzmanual.tex]:
```suggestion
This is the classical (legacy) option, with the \texttt{voltmeter} and \texttt{ammeter}. The problem is that elements are intrinsically horizontal, so they look funny if put in vertically.

File: doc/circuitikzmanual.tex [hunk 2]
```diff
@@ -1106,29 +997,123 @@ \subsection{Resistive bipoles}
...
+\end{circuitikz}
+\end{LTXexample}
+
+The anchor is positioned just on the corner of the segmented line crossing the component.
+
+\subsubsection{Resistive components customization}
+
+You can change the scale of these components (all the resistive bipoles together) by setting the key \texttt{resistors/scale}  to something different from the default \texttt{1.0}.
+
+You can change the width of these components (all the resistive bipoles together) by setting the key \texttt{resistors/width}  to something different from the default \texttt{0.8}.
```
[the remark this answers is not in the document]
Comment (Rmano) [reply doc/circuitikzmanual.tex]: > 

Why do you delete this paragraph? It's true ;-)

[the remark this answers is not in the document]
Comment (asinghvi17) [reply doc/circuitikzmanual.tex]: It was a duplicate :)

[the remark this answers is not in the document]
Comment (asinghvi17) [reply doc/circuitikzmanual.tex]: Oh, I see - I missed that one said width and the other said size.  Will make another suggestion to correct those.

File: doc/circuitikzmanual.tex [hunk 3]
```diff
@@ -1106,29 +997,123 @@ \subsection{Resistive bipoles}
...
+   \draw (0,0) to[sL, l=$L$, name=mySL] ++(3,0);
+   \node [draw, circle, inner sep=2pt] at(mySL.label) {};
+\end{circuitikz}
+\end{LTXexample}
+
+The anchor is positioned just on the corner of the segmented line crossing the component.
+
+\subsubsection{Resistive components customization}
+
+You can change the scale of these components (all the resistive bipoles together) by setting the key \texttt{resistors/scale}  to something different from the default \texttt{1.0}.
```
Comment (asinghvi17) [inline doc/circuitikzmanual.tex]:
```suggestion
You can change the scale of these components (all the resistive bipoles together) by setting the key \texttt{resistors/scale} (default \texttt{1.0}).  Similarly, you can change the widths by setting \texttt{resistors/width} (default \texttt{1.0}).

Comment (asinghvi17) [reply doc/circuitikzmanual.tex]: This along with https://github.com/circuitikz/circuitikz/pull/255/files/3c0f9d6c0c61baac8e22e4f9c83dc1eea498157c#r305571910 should work.

**Comment by asinghvi17** [comment]
@Rmano, something seems wrong with Travis here - 3 minutes seems a little low...

**Comment by Rmano** [comment]
> Edit: I see what's going on now, false alarm. If you'd like, I could make a PR to add an Appveyor file as well, which makes it easier to extract build artifacts (like the compiled manual).

Well --- I really am still a bit unconfortable with the Travis CI thing (not the test compilation, but the management of the keys to push to the pages I am not sure to understand fully). So I do not know the advantages/differences/etc with AppVeyor... I would need a tutorial PR, really ;-)

**Comment by Rmano** [comment]
A question: I think that when scaling the transistors, the arrows and the "gate dots" should not vary in size,  to stay coherent with the current arrow and the nodes/poles in the bipoles. Do you think this is correct?

File: doc/circuitikzmanual.tex [hunk 4]
```diff
@@ -1657,6 +1657,9 @@ \subsection{Other bipoles}
     \end{circuitikz}
 \end{LTXexample}
 
+\subsubsection{Misc elements customizations}\label{sec:tweak-misc}
```
Comment (asinghvi17) [inline doc/circuitikzmanual.tex]:
```suggestion
\subsubsection{Misc. element customization}\label{sec:tweak-misc}

**Comment by Rmano** [comment]
Ok, that should be the first phase of styling ready. Now all the component sizes are changeable. If you want a pre-look to the changes, they are in the manual: 

[circuitikzmanualgit.pdf](https://github.com/circuitikz/circuitikz/files/3439433/circuitikzmanualgit.pdf)

The further steps will be to add fill color and thickness (that should be much easier using the "class" I just introduced) to the style option, and a kind of "style sheet" loading and switching. 

Comments and tests are welcome.

**Comment by Rmano** [comment]
Lots of people on vacation, ¿eh? ;-) ¡good for you!

I double-checked this change; bugs can hide anywhere, but I'll commit it in order to go forward.

**Pull request merged**